sketch_oct9a.ino

```diff
@@ -227,40 +297,15 @@ void loop() {
 }
 
 void gps_loop() {
-  if (gps.getStatus() != GPS::ACTIVE) {
-    gps.initialize();
-    updateGPSLEDs();
-  }
-
-  gps.getLocation(&last_coords);
+  readGPS();
 
   const unsigned long GPS_FREQ = 18;
   delay(1000 / GPS_FREQ);
 }
 
 void radio_loop() {
   delay(1000 / RADIO_FREQ);
-
-  Frame f = createFrame(
-    last_coords,
-    last_accel,
-    last_gyro,
-    last_alt
-#if CAPSULE == 2
-    ,
-    last_voc,
-    last_humid,
-    last_temp
-#endif
-  );
-
-  // In theory, this should send the data to the radio.
-  // In practice, this is completely untested. If nothing is connected to the UART pins, however,
-  // this doesn't do anything, so it's harmless to leave in for now.
-  Serial1.write(
-    (char*)&f,
-    FRAME_SIZE()
-  );
+  sendDataToRadio();
 }
 
 #endif
```

Comment:
As a whole, I'm still not 100% sure how the code is multi-threaded. I see a few different loops going on, but next time we meet Riley I would like for you to take a few minutes and explain it to me again. Hopefully it'll stick this time.

Reply:
Discussed in person.